internal/instrumented_resp_connection/utils.go

```diff
@@ -0,0 +1,28 @@
+package instrumented_resp_connection
+
+import (
+	"fmt"
+	"strings"
+)
+
+// quoteIfHasSpaceOrEscapeSequence quotes a string if it contains escapable characters or spaces
+func quoteIfHasSpaceOrEscapeSequence(s string) string {
+	quoted := fmt.Sprintf("%q", s)
+	trimmedQuotes := strings.Trim(quoted, "\"")
+
+	// if the string does not change, no escapable chracter was present
+	if s == trimmedQuotes && !strings.Contains(s, " ") {
+		return s
+	}
+
+	return quoted
+}
+
+// quoteCLICommand applies quoteIfHasSpaceOrEscapeSequence to each item in commandWithArgs
+func quoteCLICommand(commandWithArgs []string) []string {
```

Comment:
```suggestion
func quoteCLICommand(commandWithArgs []string) string {
```

Let's make this just return a single string - similar to the library we looked at earlier. If we're quoting args, it's very unlikely we need them individually. Quoting is almost always done when you need to pass to a shell or log like we're doing - in both of those cases, the input will be concatenated before use.

Reply:
Got it!